torch/nn/modules/container.py

```diff
@@ -267,6 +383,130 @@ def extend(self, parameters):
             self.register_parameter(str(offset + i), param)
         return self
 
+    def extra_repr(self):
+        child_lines = []
+        for k, p in self._parameters.items():
+            size_str = 'x'.join(str(size) for size in p.size())
+            device_str = '' if not p.is_cuda else ' (GPU {})'.format(p.get_device())
+            parastr = 'Parameter containing: [{} of size {}{}]'.format(
+                torch.typename(p.data), size_str, device_str)
+            child_lines.append('  (' + str(k) + '): ' + parastr)
+        tmpstr = '\n'.join(child_lines)
+        return tmpstr
+
+
+class ParameterDict(Module):
```

Comment:
Actually, what if we created a single superclass that would access `self.dict` and used `self.register_elem`, which would be implemented in subclasses? This would allow us to avoid repeating all the simple methods and error checking

Reply:
It might be unnecessary because as discussed in https://github.com/pytorch/pytorch/issues/4048 there is no plan to add more such containers, there is not much duplication and the code is small enough. `ParameterList` and `ModuleList` also have duplicate code.

However, I can do it if it can get this PR merged.